[Seah Jia Yi] iP

JiaYi-Gallium369 opened the pull request:
# Gallium
> "Talk is cheap. Show me the code." ― Linus Torvalds

Gallium is more than your regular todo app. It's
- text-based
- ~~easy~~ SUPER EASY to use

All you need to do is,
1. Download the app [here] {https://github.com/JiaYi-Gallium369/ip}
2. Add your tasks
3. Let it manage your tasks! :shipit: 

And it is **FREE**!

Features:
- [x] Managing tasks
- [ ] Managing deadlines
- [ ] Reminders (coming soon)
- [ ] Managing events

Here's the `main` method:
```
    public static void main(String[] args) {
        new Gallium("data/gallium.txt").run();
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

JiaYi-Gallium369 committed:
rename bot + add greeting

JiaYi-Gallium369 committed:
add echo

JiaYi-Gallium369 committed:
add list

JiaYi-Gallium369 committed:
add mark and unmark

JiaYi-Gallium369 committed:
add class Task

JiaYi-Gallium369 committed:
add todo, deadline and event class

JiaYi-Gallium369 committed:
add ui test

JiaYi-Gallium369 committed:
add GalliumException

JiaYi-Gallium369 committed:
handle more exceptions

JiaYi-Gallium369 committed:
add delete feature

JiaYi-Gallium369 committed:
correct singular/plural for task/tasks

JiaYi-Gallium369 committed:
Add create txt file on start of bot

JiaYi-Gallium369 committed:
Add file writer to write txt file

JiaYi-Gallium369 committed:
Change Todo, Deadline and Event class constructor

JiaYi-Gallium369 committed:
Add date and time for class Deadline

JiaYi-Gallium369 committed:
Add date and time for class Event

JiaYi-Gallium369 committed:
Add 'date' command

JiaYi-Gallium369 committed:
Add Ui class and methods

JiaYi-Gallium369 committed:
Add scanner as a field in the class Ui

JiaYi-Gallium369 committed:
Add Storage class constructor and load method

JiaYi-Gallium369 committed:
Add Storage class writeFile method

JiaYi-Gallium369 committed:
Add Parser Class and methods

JiaYi-Gallium369 committed:
Change if-else into switch statements in parser.run()

JiaYi-Gallium369 committed:
Add Command classes

AngPengXuan commented on src/main/java/MarkCommand.java:
```diff
@@ -0,0 +1,25 @@
+import java.util.regex.Pattern;
+
+public class MarkCommand extends Command {
+    private String Message;
```
Should this be message instead to conform to coding standard? Variable names should be camel case.

JiaYi-Gallium369 committed:
Add files into packages and add more exception handling

AngPengXuan commented on src/main/java/Storage.java:
```diff
@@ -0,0 +1,61 @@
+import java.util.Scanner;
+
+public class Storage {
+    File filePath;
```
Should this be private? Same for the other 2 variables.

AngPengXuan commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,89 @@
+    public Command parse(String Message) {
+        while (!Message.equals(BYE)) {
+            try {
+                switch (Message) {
```
Should this switch statement be something like this instead:
```
switch (condition) {
case ABC:
    statements;
    // Fallthrough
case DEF:
    statements;
    break;
case XYZ:
    statements;
    break;
default:
    statements;
    break;
}
```

AngPengXuan commented on src/main/java/Ui.java:
```diff
@@ -0,0 +1,118 @@
+                + " in the list.");
+    }
+
+    public void showIOException(IOException e) {
```
Should this be showIoException instead?

AngPengXuan commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,76 @@
+    }
+
+    public String toString() {
+        return "[E]" + this.getStatusIcon() + this.desc + " (from: " +
```
Should the '+' be on the next line instead?

JiaYi-Gallium369 committed:
Merge branch 'add-gradle-support'

AngPengXuan commented on src/main/java/Gallium.java:
```diff
@@ -0,0 +1,39 @@
+        storage = new Storage(filePath);
+        // try {
+        taskList = new TaskList(storage.load());
+        // } catch (GalliumException e) {
```
Perhaps these comments can be removed? They should already be captured in your git history.

AngPengXuan commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,76 @@
+                e.printStackTrace();
+            }
+        } else if (description.startsWith("[E][ ] ")) {
+            String[] parts = description.split("\\[E\\]\\[ \\] ");
```
There are quite a lot of repeated code between this if and the one below, maybe can create variables to avoid repetition?

AngPengXuan commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,52 @@
+                e.printStackTrace();
+            }
+        } else if (description.startsWith("[D][ ] ")) {
+            String[] parts = description.split("\\[D\\]\\[ \\] ");
```
Both this if segment and the one below looks very similar, maybe can create another function containing the similar parts so as not to repeat so much code?

AngPengXuan commented on src/main/java/AddCommand.java:
```diff
@@ -0,0 +1,28 @@
+    }
+
+    @Override
+    public void execute(TaskList taskList, Ui ui, Storage storage) throws GalliumException {
```
Good job making use of OOP here to add tasks!

AngPengXuan commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,89 @@
+                        throw new GalliumException("OOPS!!! The description of a " + Message + " cannot be empty.");
+
+                    default:
+                        if (Message.matches(MARK + " \\d+") || Message.matches(UNMARK + " \\d+")) {
```
Maybe can add this to the cases above? Default probably shouldn't contain so many if else statements.

AngPengXuan reviewed (commented):
Overall, I found your code easy to read except the switch statement with nested if-else.  I also noted a few minor coding standard violations
too. Good job making use of OOP for Tasks and Commands!

JiaYi-Gallium369 committed:
Move files

JiaYi-Gallium369 committed:
Add testcases for all Task classes

JiaYi-Gallium369 committed:
Add testcases for Parser class

JiaYi-Gallium369 committed:
Update build.gradle

JiaYi-Gallium369 committed:
Add logo

JiaYi-Gallium369 committed:
Add jar file

JiaYi-Gallium369 committed:
add FindCommand class

JiaYi-Gallium369 committed:
Add javadocs to each class and method

JiaYi-Gallium369 committed:
Change name of variable Message to message

JiaYi-Gallium369 committed:
Change name of variable tasklist to taskList

JiaYi-Gallium369 committed:
Change indentation of switch case

JiaYi-Gallium369 committed:
Resolve merge conflicts

JiaYi-Gallium369 committed:
Resolve merge conflicts for Find Command

JiaYi-Gallium369 committed:
Add Javadocs for FindCommand class

JiaYi-Gallium369 committed:
Change indentation of switch case

JiaYi-Gallium369 committed:
Add JavaFX

JiaYi-Gallium369 committed:
Tweak the GUI

JiaYi-Gallium369 committed:
Add checkstyle

JiaYi-Gallium369 committed:
shorten parse method

JiaYi-Gallium369 committed:
Parser and Storage class: add assertions
There are no checks to verify the input of
messages before being parsed into commands. There
are no checks to ensure that files are created
successfully or tasks are correctly loaded.

Adding assertions can ensure that messages, file
paths and tasks are validated.

Let's,
* add assertions in `parse` and `returnCommand` method to
verify that no empty messages are being processed.
* add asserions to the `Storage` constructor to ensure
that the file path string is not empty and the file is
successfully created.
* add assertions in `load` method to check that task descriptions
are not empty.

JiaYi-Gallium369 committed:
Main Class: add assertions to ensure FXML and
anchor pane load successfully

The start method of Main class does not have checks
to confirm that FXML and AnchorPane are successfully loaded.
Checks are needed to spot potential issues during early stages
of execution of the app.

Let's add assertions to verify that FXMLLoader and AnchorPane is loaded
correctly.

These checks can help to debug and prevent further errors during the
loading of the UI.

JiaYi-Gallium369 committed:
Gallium class: Add Assertions to ensure every
field is initialized

Gallium class initializes the most neccessary
components, such as Ui, Storage and TaskList.
There are no checks to confirm that these
elements are properly set up.

Let's:
* add assertions to the Gallium constructor to
check if each field is being initialized.
* add assertions to check if getOutput of ui is
not an empty string.

JiaYi-Gallium369 committed:
Reduce method length for parse method in Parse class

JiaYi-Gallium369 committed:
Reduce length for Deadline constructor

JiaYi-Gallium369 committed:
Modify methods handling Deadline class to be uniform with Todo and Event class

JiaYi-Gallium369 committed:
Add IllegalArgumentException handling for Deadline constructor

JiaYi-Gallium369 committed:
Reduce length for Event constructor

JiaYi-Gallium369 committed:
Trim trailing spaces of the message to reduce the number of cases being handled in parse method

JiaYi-Gallium369 committed:
Add methods to shorten the parse method

JiaYi-Gallium369 committed:
Reduce arrowhead method for execute method

JiaYi-Gallium369 committed:
Remove unnecessary method

JiaYi-Gallium369 committed:
Remove magic string

JiaYi-Gallium369 committed:
Add more methods to shorten the Event and Deadline constructor

JiaYi-Gallium369 committed:
Merge pull request #2 from JiaYi-Gallium369/branch-A-Assertions

Branch a assertions

JiaYi-Gallium369 committed:
Resolve merge conflicts

JiaYi-Gallium369 committed:
Merge pull request #3 from JiaYi-Gallium369/branch-A-CodeQuality

Branch a code quality

JiaYi-Gallium369 committed:
Add EditCommand class

JiaYi-Gallium369 committed:
Add changes to parse method to handle the edit command

JiaYi-Gallium369 committed:
Add a EditCommand variable in Gallium class to enable handle user input twice

JiaYi-Gallium369 committed:
Add more messages to handle edit command in Ui class

JiaYi-Gallium369 committed:
Add methods in Task class to handle the editing of its fields

JiaYi-Gallium369 committed:
Event class: Implement error handling for cases where from is later than to

JiaYi-Gallium369 committed:
Event class: Remove unused imports

JiaYi-Gallium369 committed:
Change picture of DaUser

JiaYi-Gallium369 committed:
Add icon to title bar

JiaYi-Gallium369 committed:
Remove DaGallium image background

JiaYi-Gallium369 committed:
Change colors of the background

JiaYi-Gallium369 committed:
Add padding and margin for dialog box

JiaYi-Gallium369 committed:
Edit taskList in gallium.txt

JiaYi-Gallium369 committed:
Add Ui.png

JiaYi-Gallium369 committed:
Add README

JiaYi-Gallium369 committed:
Edit README

JiaYi-Gallium369 committed:
Add Ui.png

JiaYi-Gallium369 committed:
Add Ui.png

JiaYi-Gallium369 committed:
Add personality for Gallium

JiaYi-Gallium369 committed:
Change personality for Gallium

JiaYi-Gallium369 committed:
Change more personality for Gallium

JiaYi-Gallium369 committed:
Add handling incomplete deadline details error

JiaYi-Gallium369 committed:
Fix line spacing in README

JiaYi-Gallium369 committed:
Add error handling for invalid add command

JiaYi-Gallium369 committed:
DateCommand Class: Skip todo task

JiaYi-Gallium369 committed:
EditCommand class: Handle non-number inputs

JiaYi-Gallium369 committed:
Add error handling for invalid commands

JiaYi-Gallium369 committed:
Add quick start in README

JiaYi-Gallium369 committed:
Fix quick start header

JiaYi-Gallium369 committed:
Add storage details into README

JiaYi-Gallium369 committed:
Add help command

JiaYi-Gallium369 committed:
Set to auto exit the app if isExit is true

JiaYi-Gallium369 committed:
Add HelpCommand into README

JiaYi-Gallium369 committed:
Change Ui.png
